extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -141,8 +142,14 @@ public KafkaStreams getStreams() {
 
     private void waitForTopicsToBeCreated(Set<String> topicsToAwait, String bootstrapServersConfig)
             throws InterruptedException {
-        Map<String, Object> config = new HashMap<>();
+        final Map<String, Object> config = new HashMap<>();
         config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServersConfig);
+        // include other AdminClientConfig(s) that have been configured
+        for (final String knownAdminClientConfig : AdminClientConfig.configNames()) {
```

Comment:
Ok, I think that goes into the right direction. Only that it should pull the properties from the `kafka-streams` namespace, or better yet `kafka-streams.admin`; i.e. it should work to provide the properties like so:

    kafka-streams.admin.security.protocol=SSL
    ...

Reply:
Hi @gunnarmorling, at least in the context of this specific issue which prompted me to look into it, the configurations that are relevant are certain "common" configurations (SSL etc) which are (typically) applicable to consumers, producers and admin client(s) of Kafka streams.
Do you expect the values for these configs to be different in context of different uses (consumer vs producer vs admin client)? In other words, at least for the current configs, do we need an additional kafka-streams.admin.. prefix? As it is, this current change in PR makes it a point that only those properties that are known/recognized by AdminClient will be passed to it.
I'm open to making a new namespace as you suggest, but just want to be sure we need it.

Reply:
Just noticed one part in your comment:
> Ok, I think that goes into the right direction. Only that it should pull the properties from the kafka-streams namespace

The `properties` member variable that you see in there, in the `KafkaStreamsTopologyManager` is actually built off the "kafka-streams" namespace. That part is already handled here  https://github.com/quarkusio/quarkus/blob/master/extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java#L142. So the contents of the `properties` member are those properties that begin with `kafka-streams.` prefix, with their prefixes removed.

Reply:
@jaikiran, ok, so it seems we agree there should be the "kafka-streams" prefix, right? Unless I'm mistaken, that's not yet part of the PR?

About the "admin" sub-namespace, the idea was to resemble what you'd two when directly configuing Kafka Streams, where you'd also use `StreamsConfig#adminClientPrefix()` and related methods to prefix these values (see [here](https://docs.confluent.io/current/streams/developer-guide/config-streams.html#naming)). So to specify any value for the KStreams *internal*  admin client, the `kafka-streams.admin.` prefix would have to be used anyways (and `kafka-streams.` would be removed before passing on the value).

Now the question is is there the need for specifying different values for our own admin client here than for the KStreams internal one. If not, I'd argue, the `kafka-streams.admin.` prefix should be used for both.

Reply:
Hi @gunnarmorling, I looked at the link you pasted in your comment and I see what you mean. So what I have now done is updated this PR to change the way I pass the configs to the `AdminClient`. What it does now (snippet):
```
final Map<String, Object> config = new HashMap<>();
        config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServersConfig);
        // include other AdminClientConfig(s) that have been configured
        for (final String knownAdminClientConfig : AdminClientConfig.configNames()) {
            // give preference to admin.<propname> first
            if (properties.containsKey("admin." + knownAdminClientConfig)) {
                config.put(knownAdminClientConfig, properties.get("admin." + knownAdminClientConfig));
            } else if (properties.containsKey(knownAdminClientConfig)) {
                config.put(knownAdminClientConfig, properties.get(knownAdminClientConfig));
            }
        }
```
So it first checks if the `properties` has any `admin.<propname>` (note the `kafka-streams.` prefix is already stripped so no need to check for that) and if it finds such a property, it uses it. Else it checks if any property that's recognized by `AdminClientConfig` exists and if it finds it then uses it. This way, users can now use either (an example):
```
kafka-streams.admin.security.protocol=SSL
```
or if they want to share the config value for all (consumers, producer and admin client), then:

```
kafka-streams.security.protocol=SSL
```
and the `AdminClient` that's constructed in this `KafkaTopologyManager` will honour it correctly.

Do you think this is fine?

Reply:
Yes, I think that makes sense. I'd only suggest to reference `org.apache.kafka.streams.StreamsConfig.ADMIN_CLIENT_PREFIX` instead of specifying "admin." literally. And it would be nice to have a test, but I'm not sure how it could look like, other than changing the existing integration tests to make use of SSL. Would be nice if you're up for it, but no blocker IMHO.

Reply:
>> I'd only suggest to reference org.apache.kafka.streams.StreamsConfig.ADMIN_CLIENT_PREFIX instead of specifying "admin." literally.

That makes sense. Done - I have updated this PR to make that change.

>> And it would be nice to have a test, but I'm not sure how it could look like, other than changing the existing integration tests to make use of SSL. Would be nice if you're up for it, but no blocker IMHO.

Agreed. I have been adding tests wherever possible to my PRs to this project. But for this one, I still need to understand and read up on the test infrastructure a bit more to be able to add anything useful. I probably will get to it at some point, but I don't think I will be able to do it in the next couple of days.

Reply:
> but I don't think I will be able to do it in the next couple of days.

Yes, I think that's fine. To make the change eventually, you'd have to look into https://github.com/quarkusio/quarkus/blob/master/integration-tests/kafka/src/test/java/io/quarkus/it/kafka/KafkaTestResource.java, which is where the Kafka broker used for testing is started. This would have to be changed to use SSL; then the tests using that broker would have to be altered to specify the SSL options when connecting.

Reply:
Thank you for that input @gunnarmorling. That definitely helps. I'll go through it and come up with a test case in the next few days.